### ISSUE TITLE
RFID Badges for team members

### ISSUE THREAD

**Issue by RajGandhi** [FRCTeam3206/Programming#8]
It sounds like Daniel has the RFID to Arduino communication functional.  There are a myriad of outputs possible.  A challenge will be constraining the many possibilities down to something that can be implemented.

The process can be an informal combination of sketching what the output should look like and requirements.  It probably isn't value-added to write formal requirements.  Here's a smattering of ideas:

Output (initially pick just one):
- Numeric-only hours participation: boys, girls, mentors
- Web page with bar chart showing monthly participation of: boys, girls, mentors
- Web page showing who is currently logged-in

Possible requirements:
- Actual time (need ethernet shield)
- Custom record
- Non-volitile data storage
- Per person, accumulate time & keep discrete records of attendance
- confirmation beep or light to log-in/out
- communicates with a corresponding Java app

Don't worry about:
- school firewalls; assume it is good enough to locally host (old team laptop)
- if we need an Arduino Ethernet shield let me know and we'll get one (or more other stuff)

Basically, for next meeting (Thursday) identify plan and obstacles (e.g. learn to retrieve time with Ethernet shield, use Records, use a Linked List, etc.) so that we can continue making forward progress.  Update this GitHub thread to document progress; archive code (snippets) on GitHub.  Try to do it in a way that someone with similar skills could follow and replicate.

**Comment by rnrknott** [comment]
There is a wifi shield available for spark fun that might be good to add  also discussed using another one of these as a gimmick in the pit during competition  hand out rfid enabled business cards

**Comment by Daniel-Thalman** [comment]
I will figure out by next tuesday if the wifi sheild is what we want

**Comment by RajGandhi** [comment]
Re assigned issue to Steve because I'm trying to figure-out how to use GitHub better.  I'll copy the meat of the original issue to Steve via email.

**Comment by Daniel-Thalman** [comment]
After I started to use the arduino DateTime library I found that the best way to set it up for our purpose would be to use a wifi shield.  To be more specific it would be tremendously helpful for syncing the time on the arduino which is essential for the RFid tags.  I found a shield from adafruit that includes a library.  Also to make this work we would also need stacking pins also from adafruit.

Shield:
https://www.adafruit.com/products/1491

Stacking pins:
https://www.adafruit.com/products/85

**Comment by RajGandhi** [comment]
Ok, I'll plan to start the order process early next week. There are several
little item on the wish list also.

**Comment by RajGandhi** [comment]
What is the status of the RFID cards from Amazon? Are they compatible and
working?

**Comment by Daniel-Thalman** [comment]
Yew they will work

**Comment by RajGandhi** [comment]
Ordering issues... the request was for an Arduino WiFi shield and stacking
headers for use with the RFID reader.  The issue and alternative are listed
briefly below.  I'm planning to place the order with SparkFun.  Let me know
if you have objections.

Raj

​​The preferred Wifi Shield (1491) is out of stock.  (A note on the site
says this board is being phased-out in favor of the 101.)
Here's a list of more
https://www.adafruit.com/category/851?q=WiFi%20Shield& wifi devices.  The
101 is $10 more and doesn't have the SD card or proto area.
Another note in the 101 Wifi shield said that it has "basic" compatibility
with the Uno but is designed for the Arduino Zero ($55).

​
Different vendor, SparkFun:
- $14.95 Wifi Shield https://www.sparkfun.com/products/13287
- $39.95 or WiFi Shield with faster data
  https://www.sparkfun.com/products/12071

I'm inclined to go with the cheap SparkFun offering.

SparkFun has the CAN shield but Adafruit does not.
SparkFun has the IO expander for $13 and Adafruit does not.
SparkFun has a Arduino compatible for $20
https://www.sparkfun.com/products/12757 (regular price genuine Uno is
$25).  Yes, I know there's a risk with non-genuine parts but I think the
risk is worth it to try.

**Comment by Daniel-Thalman** [comment]
The board seems to be fine since it is a redesigned board not a knock off also the wifi sheild dosn't seem to have a library for it. Tell me if im wrong_

**Comment by RajGandhi** [comment]
WiFi Shield https://www.sparkfun.com/products/13287 GitHub library
https://github.com/sparkfun/SparkFun_ESP8266_AT_Arduino_Library.  I
didn't look closely at the library for issues.

**Comment by Daniel-Thalman** [comment]
Thanks sounds good

**Comment by steeve-o** [comment]
https://data.sparkfun.com/

one option for data streaming - will be public. 

Pros: anyone, anywhere, from any device with internet access and correct URL can see who is logged into the system

Cons: anyone, anywhere, from any device with internet access and correct URL can see who is logged into the system

**Comment by steeve-o** [comment]
How to install Arduino library for SparkFun WiFi shield:

https://learn.sparkfun.com/tutorials/esp8266-wifi-shield-hookup-guide#installing-the-esp8266-at-library

**Comment by ghost** [comment]
I was reading a page about programming Bluetooth on Android and I found a comment at the bottom.  Team 4488 uses a tablet and everyone has a QR code assigned to them and they scan it as a way to track attendance.  Probably cheaper than RFID and all you need is a camera App.  I believe they shared their code.  Thought you all might be interested.

Here's the post text:

"Also, another thought – it just so happens that I am a volunteer engineering mentor for a FIRST Robotics team (#4488). One of the students is implementing an attendance app in App Inventor.
Each student has a Team #4488 ID card they are supposed to have with them during each robotics meeting and event. On the card we are printing a QR Code with the student’s ID#. The attendance app does a QR code scan of the student IDs. This does, of course, require the students to sequentially walk by the Nexus 7 tablet we have out at the beginning of meetings.
I imagine we will be making this code public, once its ready, as our team tries to help out other robotics teams as much as possible. Even if you are not doing robots, you’d be welcome to the code!
If you want to write your own, I have sample code to scan QR codes available in this tutorial
http://appinventor.pevest.com/?p=1086
Ed"

Here's the page: http://appinventor.pevest.com/?p=520
